docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream utilities for downloading files, it is time to introduce you to additional *upstream* utilities. These are utilities that publish files to the Hub from your library. This guide will show you how to:
```

Comment:
Again I would suggest to avoid assuming readers are using documentation linearly. If mentioning other sections, please link to it (such as downstream here)